rest/test/db/CatapultDb_spec.js

```diff
@@ -1512,16 +1518,657 @@ describe('catapult db', () => {
+									expect(element.id).to.deep.equal(createObjectId(((pageNumber - 1) * 10) + i));
+								});
+								expect(page.paging).to.deep.equal({
+									totalEntries: 32, pageNumber, pageSize: 10
+								});
+							}
+						);
+					});
+				};
+
+				runPageNumberTest(1, 10);
+				runPageNumberTest(2, 10);
+				runPageNumberTest(3, 10);
+				runPageNumberTest(4, 2);
+				runPageNumberTest(5, 0);
+			});
+		});
+	});
+
+	describe('transactions', () => {
+		const testPublicKey = {
+			one: test.random.publicKey(),
+			two: test.random.publicKey(),
+			three: test.random.publicKey()
+		};
+		const testAddress = {
+			one: keyToAddress(testPublicKey.one),
+			two: keyToAddress(testPublicKey.two),
+			three: keyToAddress(testPublicKey.three)
+		};
+
+		const paginationOptions = {
+			pageSize: 10,
+			pageNumber: 1
+		};
+
+		const { createObjectId } = test.db;
+
+		const createTransaction = (objectId, addresses, height, signerPublicKey, recipientAddress, type) => ({
+			_id: createObjectId(objectId),
+			meta: {
+				height,
+				addresses: addresses.map(a => new Binary(a))
+			},
+			transaction: {
+				signerPublicKey,
+				recipientAddress,
+				type
+			}
+		});
+
+		const createInnerTransaction = (objectId, aggregateId, signerPublicKey, recipientAddress, type) => ({
+			_id: createObjectId(objectId),
+			meta: { aggregateId: createObjectId(aggregateId) },
+			transaction: {
+				signerPublicKey,
+				recipientAddress,
+				type
+			}
+		});
+
+		it('returns expected structure', () => {
+			// Arrange:
+			const dbTransactions = [
+				createTransaction(10, [testAddress.one], 123, testPublicKey.one, testAddress.two, EntityType.transfer)
+			];
+
+			// Act + Assert:
+			return runDbTest(
+				{ transactions: dbTransactions },
+				db => db.transactions({}, paginationOptions),
+				transactionsPage => {
+					const expected_keys = ['meta', 'transaction', 'id'];
+					expect(Object.keys(transactionsPage.data[0]).sort()).to.deep.equal(expected_keys.sort());
+				}
+			);
+		});
+
+		it('renames _id to id', () => {
+			// Arrange:
+			const dbTransactions = [
+				createTransaction(123, [testAddress.one], 1)
+			];
+
+			// Act + Assert:
+			return runDbTest(
+				{ transactions: dbTransactions },
+				db => db.transactions({}, paginationOptions),
+				transactionsPage => {
+					expect(transactionsPage.data[0].id).to.deep.equal(createObjectId(123));
+					expect(transactionsPage.data[0]._id).to.equal(undefined);
+				}
+			);
+		});
+
+		it('does not expose private meta.addresses field', () => {
+			// Arrange:
+			const dbTransactions = [
+				createTransaction(10, [testAddress.one], 1, 0, 0, 0)
+			];
+
+			// Act + Assert:
+			return runDbTest(
+				{ transactions: dbTransactions },
+				db => db.transactions({}, paginationOptions),
+				transactionsPage => {
+					expect(transactionsPage.data.length).to.equal(1);
+					expect(undefined === transactionsPage.data[0].meta.addresses).to.equal(true);
```

Comment:
i think `to.be.undefined` exists, otherwise change to `to.equal(undefined)` just like you did above - review this in the whole document